### ISSUE TITLE
High CPU usage for DNS Client

### ISSUE THREAD

**Issue by TheCakeIsNaOH** [tailscale/tailscale#3274]
### What is the issue?

I am seeing high CPU usage with DNS Client. 

There seems to bursts of a couple of seconds when connecting to tailscale, and at intervals when connected. The bursts of CPU usage are at anywhere from 5% to 20% CPU usage (on a high core count CPU).

### Steps to reproduce

Just disconnect then reconnect to tailscale from the tray menu.

### OS

Windows

### OS version

Windows 10 20H2

### Tailscale version

1.16.2

### Bug report

BUG-87e1c7cd03306fbe2a218c70c71d9e2fa563e71c2ce09f4c4d6ef3c505828925-20211106235836Z-a88eb23bf4a32823

**Comment by DentonGentry** [comment]
```
2021-11-06 18:54:10.5497314 -0500 -0500: 2021/11/06 18:54:10 dns: error: context deadline exceeded
2021-11-06 18:54:10.5497314 -0500 -0500: 2021/11/06 18:54:10 dns: error: context deadline exceeded
2021-11-06 18:54:13.3057212 -0500 -0500: 2021/11/06 18:54:13 open-conn-track: timeout opening (TCP 100.x.y.z:1974 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:14.5028361 -0500 -0500: 2021/11/06 18:54:14 open-conn-track: timeout opening (TCP 100.x.y.z:8320 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:14.6988789 -0500 -0500: 2021/11/06 18:54:14 open-conn-track: timeout opening (TCP 100.w.x.y.z:30277 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:14.9096217 -0500 -0500: 2021/11/06 18:54:14 open-conn-track: timeout opening (TCP 100.x.y.z:1025 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:15.119818 -0500 -0500: 2021/11/06 18:54:15 open-conn-track: timeout opening (TCP 100.x.y.z:6543 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:15.3468199 -0500 -0500: 2021/11/06 18:54:15 open-conn-track: timeout opening (TCP 100.x.y.z:1028 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:16.0631396 -0500 -0500: 2021/11/06 18:54:16 Accept: UDP{100.x.y.z:63180 > 100.q.r.s:53} 60 ok out
2021-11-06 18:54:17.526934 -0500 -0500: 2021/11/06 18:54:17 open-conn-track: timeout opening (TCP 100.x.y.z:8320 => 100.100.100.100:53); peerForIP: exit node found but notenabled
2021-11-06 18:54:17.7243656 -0500 -0500: 2021/11/06 18:54:17 open-conn-track: timeout opening (TCP 100.x.y.z:30277 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:17.9364855 -0500 -0500: 2021/11/06 18:54:17 open-conn-track: timeout opening (TCP 100.x.y.z:1025 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:18.1492328 -0500 -0500: 2021/11/06 18:54:18 open-conn-track: timeout opening (TCP 100.x.y.z:6543 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:18.3616092 -0500 -0500: 2021/11/06 18:54:18 open-conn-track: timeout opening (TCP 100.x.y.z:1028 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:25.3182182 -0500 -0500: 2021/11/06 18:54:25 open-conn-track: timeout opening (TCP 100.x.y.z:1974 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:26.242225 -0500 -0500: 2021/11/06 18:54:26 Accept: UDP{100.x.y.z:58517 > 100.q.r.s:53} 60 ok out
2021-11-06 18:54:29.5281775 -0500 -0500: 2021/11/06 18:54:29 open-conn-track: timeout opening (TCP 100.x.y.z:8320 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:29.724822 -0500 -0500: 2021/11/06 18:54:29 open-conn-track: timeout opening (TCP 100.x.y.z:30277 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:29.9374965 -0500 -0500: 2021/11/06 18:54:29 open-conn-track: timeout opening (TCP 100.x.y.z:1025 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:30.1508615 -0500 -0500: 2021/11/06 18:54:30 open-conn-track: timeout opening (TCP 100.x.y.z:6543 => 100.100.100.100:53); peerForIP: exit node found but not enabled
2021-11-06 18:54:30.3620271 -0500 -0500: 2021/11/06 18:54:30 open-conn-track: timeout opening (TCP 100.x.y.z:1028 => 100.100.100.100:53); peerForIP: exit node found but not enabled
```

The `exit node found but not enabled` message is misleading, that is a bug which can happen with 1.16.0. I note two devices on your tailnet running 1.16.0, and I suspect that one of them it involved in this issue. Can you check https://login.tailscale.com/admin/machines to identify those two devices and update them to 1.16.2?

**Comment by TheCakeIsNaOH** [comment]
I have updated all the devices in my network to 1.16.2 and am still seeing the same behavior.

**Comment by DentonGentry** [comment]
I think this may be an instance of [#5879](https://github.com/tailscale/tailscale/issues/5879), which was fixed in Tailscale 1.32.1.

Barring further comment, we'll expect to close this during the next regular bug scrub.

**Comment by TheCakeIsNaOH** [comment]
I'm no longer seeing the random spikes of CPU
